### DISCUSSION TITLE
Does urql caching work even in next.js's react server components implementation?

### DISCUSSION THREAD

**Discussion by xoldd** [urql-graphql/urql#3362] [Q&A]
@kitten @JoviDeCroock  The code snippet from urql docs:-
```typescript
// app/page.tsx
import React from 'react';
import Head from 'next/head';
import { cacheExchange, createClient, fetchExchange, gql } from '@urql/core';
import { registerUrql } from '@urql/next/rsc';

const makeClient = () => {
  return createClient({
    url: 'https://trygql.formidable.dev/graphql/basic-pokedex',
    exchanges: [cacheExchange, fetchExchange],
  });
};

const { getClient } = registerUrql(makeClient);

export default async function Home() {
  const result = await getClient().query(PokemonsQuery, {});
  return (
    <main>
      <h1>This is rendered as part of an RSC</h1>
      <ul>
        {result.data.pokemons.map((x: any) => (
          <li key={x.id}>{x.name}</li>
        ))}
      </ul>
    </main>
  );
}
```
Here the initialized urql client is making use of **cacheExchange** which works on the document cache based caching approach. Does this cache work and get reused throughout the application?

**Comment by JoviDeCroock** [comment]
It should, why, have you noticed something else?

**Reply by xoldd** [reply]
@JoviDeCroock No I just wanted to confirm. This is confusing because it is server side cache instead of client side. Anyway this is kind of weird, I wonder if the cache only works if next.js application is deployed on vercel because I don't think all serverless platforms would persist cache like that?

Also, does the server side cache invalidation work globally? Like if a cached entity is invalidated, would it get invalidated from each page within the application? Would graphcache exchange work server side as well? The server side cache also has pitfalls when cached data is highly user specific.

I think it'd be great to include all this on urql docs.

**Reply by JoviDeCroock** [reply]
I would like to point out that it's quite impolite to start tagging people for questions 😅 we are here in our free time and tagging folks expresses an urgency... we all still have a job and this is quite annoying...

**Reply by xoldd** [reply]
Uhhh sure. Whenever you have time.

**Comment by frantisek-heca** [comment]
Well, what's the answer to this question, please?
Is it really so, that both document cache and graphcache are working even on the server side (for example getServerSideProps or rsc)?
Does it really construct "in memory" (Map) cache on the server? If it is in memory, it's only viable during a single request/response, right?
What's the point of this cache (if it's really active on the server)? Is the main (only?) reason to deduplicate data fetches during the single request/response cycle?
